code/game/objects/items/crayons.dm

```diff
@@ -8,71 +14,203 @@
+	// Makes crayons identifiable in things like grinders
+	if(name == "crayon")
+		name = "[item_color] crayon"
+
 	if(config)
 		if(config.mutant_races == 1)
 			graffiti |= "antilizard"
 			graffiti |= "prolizard"
 
-/obj/item/toy/crayon/initialize()
-	if(config.mutant_races == 1)
-		graffiti |= "antilizard"
-		graffiti |= "prolizard"
-
-/obj/item/toy/crayon/attack_self(mob/living/user)
-	update_window(user)
-
-/obj/item/toy/crayon/proc/update_window(mob/living/user)
-	dat += "<center><h2>Currently selected: [drawtype]</h2><br>"
-	dat += "<a href='?src=\ref[src];type=random_letter'>Random letter</a><a href='?src=\ref[src];type=letter'>Pick letter/number</a>"
-	dat += "<a href='?src=\ref[src];buffer=1'>Write</a>"
-	dat += "<hr>"
-	dat += "<h3>Runes:</h3><br>"
-	dat += "<a href='?src=\ref[src];type=random_rune'>Random rune</a>"
-	for(var/i = 1; i <= 6; i++)
-		dat += "<a href='?src=\ref[src];type=rune[i]'>Rune[i]</a>"
-		if(!((i + 1) % 3)) //3 buttons in a row
-			dat += "<br>"
-	dat += "<hr>"
-	graffiti.Find()
-	dat += "<h3>Graffiti:</h3><br>"
-	dat += "<a href='?src=\ref[src];type=random_graffiti'>Random graffiti</a>"
-	var/c = 1
-	for(var/T in graffiti)
-		dat += "<a href='?src=\ref[src];type=[T]'>[T]</a>"
-		if(!((c + 1) % 3)) //3 buttons in a row
-			dat += "<br>"
-		c++
-	dat += "<hr>"
-	var/datum/browser/popup = new(user, "crayon", name, 300, 500)
-	popup.set_content(dat)
-	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
-	popup.open()
-	dat = ""
+	all_drawables = graffiti + letters + numerals + oriented + runes
+	drawtype = pick(all_drawables)
+
+	refill()
+
+/obj/item/toy/crayon/Destroy()
+	if(reagents)
+		qdel(reagents)
+	. = ..()
+
+/obj/item/toy/crayon/proc/refill()
+	if(charges == -1)
+		charges_left = 100
+	else
+		charges_left = charges
+
+	if(!reagents)
+		create_reagents(charges_left * volume_multiplier)
+	reagents.clear_reagents()
+
+	var/total_weight = 0
+	for(var/key in reagent_contents)
+		total_weight += reagent_contents[key]
+
+	var/units_per_weight = reagents.maximum_volume / total_weight
+	for(var/reagent in reagent_contents)
+		var/weight = reagent_contents[reagent]
+		var/amount = weight * units_per_weight
+		reagents.add_reagent(reagent, amount)
+
+/obj/item/toy/crayon/proc/use_charges(amount)
+	// Returns number of charges actually used
+	if(charges == -1)
+		. = amount
+		refill()
+	else
+		. = min(charges_left, amount)
+		charges_left -= .
+
+/obj/item/toy/crayon/proc/check_empty(mob/user)
+	// When eating a crayon, check_empty() can be called twice producing
+	// two messages unless we check for being deleted first
+	if(!qdeleted(src))
+		. = TRUE
+
+	. = FALSE
+	// -1 is unlimited charges
+	if(charges == -1)
+		. = FALSE
+	else if(!charges_left)
+		user << "<span class='warning'>There is no more of [src.name] \
+			left!</span>"
+		if(self_contained)
+			qdel(src)
+		. = TRUE
+
+/obj/item/toy/crayon/ui_interact(mob/user, ui_key = "main", \
```

Comment:
yes please move this back to one line

I don't' mind it in comments or strings, but this I don't like, it makes it difficult to grep for procs